### ISSUE TITLE
how can i add .js extension to module specifier in src folder

### ISSUE THREAD

**Issue by iyjian** [GervinFung/ts-add-js-extension#61]
i am going to transform a ts project to esm, so i have to add .js extension to each existing import statement, and after install `ts-add-js-extension`:

```bash
pnpm i -D ts-add-js-extension
```
i execute the following command:

```bash
 npx ts-add-js-extension add --dir=src --include=src --showchanges=true
```

but nothing changes in my project srouce file, could u have a look at this situation?

**Comment by GervinFung** [comment]
1st of all, you can omit `add`, that's only for backward compatibility, I'll update the README ASAP

You can also omit `--include=src` because you already specify `--dir=src`

**Comment by iyjian** [comment]
```bash
git clone git@gitlab.com:iyjian/nestjs_api_builder.git && cd nestjs_api_builder/api && pnpm i

npx ts-add-js-extension  
```

**Comment by iyjian** [comment]
i am not sure if you can access my gitpod snapshot workspace, i installed all dependencies in that snapshot.


https://gitpod.io#snapshot/45920d49-873a-43b4-85bf-63bff60bbfae

**Comment by GervinFung** [comment]
> ```shell
> git clone git@gitlab.com:iyjian/nestjs_api_builder.git && cd nestjs_api_builder/api && pnpm i
> 
> npx ts-add-js-extension  
> ```

Thanks for the quick reply! Is ur repo private?

**Comment by iyjian** [comment]
it should be a public repo

you can also clone this mirror:

```bash
git clone https://github.com/iyjian/nestjs_api_builder.git && cd nestjs_api_builder/api && pnpm i

npx ts-add-js-extension 
```

**Comment by GervinFung** [comment]
I cloned it, the reason nothing change in ur `src` folder is that - `ts-add-js-extension` were meant to run on JavaScript files, which means, when `tsc` transpile TypeScript code in `src` to ESM JavaScript code in `dist`, and only then u write `ts-add-js-extension --dir=dist`

**Comment by iyjian** [comment]
ah, i see

I'd like to transform my ts project to esm module project, I want add .js extension to my  source file explicitly, how can i use this tool to do this. or any clue, i can pr for this feature, since I have a bounch of projects to transform.

**Comment by GervinFung** [comment]
> ah, i see
> 
> I'd like to transform my ts project to esm module project, I want add .js extension to my source file explicitly, how can i use this tool to do this. or any clue, i can pr for this feature, since I have a bounch of projects to transform.

U can transform to ESM without modifying `src` folders, in fact it's discouraged. The philosophy of this tool is to not modify source folder, because u r importing/exporting TypeScript file, therefore the file extension shouldn't be `.js`, it should be `.js` after source code is transpiled to JavaScript.

TLDR; source code shouldn't be referencing build artifacts

Can I know any reason u wanna add JavaScript file extension to ur source files?

**Comment by iyjian** [comment]
after i read document of [ECMAScript Modules in Node.js](https://www.typescriptlang.org/docs/handbook/esm-node.html) , I thought put `.js` extension in typescript source code is recommended way, since this way support both commonjs and esm. 

also, I found many opensource projects are using this way.

ex. https://github.com/wechaty/wechaty/blob/main/src/pure-functions/mod.ts

**Comment by GervinFung** [comment]
@iyjian how is things going? Did I answer your question?
